subprojects/core/src/main/java/org/gradle/internal/build/NestedRootBuildRunner.java

```diff
@@ -38,12 +39,13 @@ public static void runNestedRootBuild(String buildName, StartParameterInternal s
     }
 
     public static NestedBuildTree createNestedBuildTree(@Nullable String buildName, StartParameterInternal startParameter, ServiceRegistry services) {
+        BuildInvocationScopeId buildInvocationScopeId = services.get(BuildInvocationScopeId.class);
```

Comment:
Do we need to do here:
```
BuildInvocationScopeId buildInvocationScopeId = new BuildInvocationScopeId(UniqueId.generate());
```
due to the comment that `GradleBuild`task should have a different id?

Reply:
No, the idea is that builds triggered by the `GradleBuild` tasks should inherit the invocation ID from the build the parent build (where the `GradleBuild` task is running). This is checked specifically by `ScopeIdsIntegrationTest`.